commands/cli-config.go

```diff
@@ -684,16 +684,25 @@ func handleCliConfigInitCmd(vcli vaultcli.CLI, args []string) int {
 
 	// Authentication type and authentication data.
 	authType := viper.GetString(cst.AuthType)
+	clientID := ""
 	if authType == "" {
 		authType, err = promptAuthType()
 		if err != nil {
 			vcli.Out().WriteResponse(nil, errors.New(err))
 			return 1
 		}
 		viper.Set(cst.AuthType, authType)
+		if authType == string(auth.FederatedAzure) {
```

Comment:
Please add another case into the switch below and move prompting for client ID to that case.
This `if`-block is only for getting authentication type if it was not provided via flags.

Reply:
good catch, thanks. This modification is done